container.go

```diff
@@ -38,16 +38,16 @@ func handleContainer(
 		return
 	}
 
-	hostname := extractHostname(ctx, container)
+	hostnames := extractHostnames(ctx, container)
 	services := extractServices(ctx, container)
 
-	if hostname != "" {
+	for _, hostname := range hostnames {
 		hostname = rewriteHostname(hostname)
 		addToDNS(eg, hostname, ips, services, container.Name[1:], true)
```

Comment:
The final true parameter indicates if the hostname should be publish for service discovery.

We can only publish it as a service once. So you would have to set the parameter to true only on the first iteration. Or figure out how to publish it multiple times :)